tests/utils.py

```diff
@@ -2,10 +2,16 @@
 # See LICENSE file for licensing details.
 
 import os
+import subprocess
 
 import nbformat
 
 
+def install_python_requirements(requirements_file: str = "requirements.txt", *args, **kwargs):
+    """Install Python dependencies specified in the provided requirements file."""
+    subprocess.run(["pip", "install", "-r", requirements_file])
```

Comment:
Can we assume that there will be an exception if the exit code of this is != 0?

Reply:
If we really want to raise an exception there I could add a `check=True`, but I was thinking of this as a best-effort thing. For instance, `pip install` could fail to resolve package versions like [this](https://github.com/canonical/charmed-kubeflow-uats/blob/main/tests/notebooks/minio-integration/minio-integration.ipynb), but that won't necessarily affect the notebook execution. Do we want to block on `pip install`?

Reply:
I'm expecting that we'll have install errors BTW as long as we use this base image (https://github.com/canonical/charmed-kubeflow-uats/issues/3) and have no env isolation (https://github.com/canonical/charmed-kubeflow-uats/issues/9).

Reply:
I wonder if by not raising or noticing we are not taking care of real issues. Let's say that we are unable to install kfp (because of whatever reason), and we proceed with the notebook execution.

Reply:
Do you have any ideas about how we could approach this? I'm ok with the approach that, well,  if nothing breaks further down, it's actually fine.

Reply:
That's fair.